app/src/main/java/com/example/tour_a_versity_v3/ui/MapsActivity.kt

```diff
@@ -35,39 +38,61 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnInfoWi
         this.map = googleMap
         map.mapType = GoogleMap.MAP_TYPE_HYBRID
         map.setOnInfoWindowClickListener(this)
-        //set values for TUC
-        val tucLat = 39.13175
-        val tucLng = -84.51774
-        val zoomLevel = 17f
-        //create LatLng object for TUC
-        val tucLatLng = LatLng(tucLat, tucLng)
+
         //move camera to TUC & add marker
-        map.moveCamera(CameraUpdateFactory.newLatLngZoom(tucLatLng, zoomLevel))
-        tucMarker = map.addMarker(MarkerOptions()
-                .position(tucLatLng)
-                .title("TUC")
-                .snippet("Tangeman University Center")
-        )
+        moveCameraToTuc()
+        tucMarker = getTucMarker()
 
         //create another marker for CECH
-        val cechLat = 39.130315
-        val cechLng = -84.518680
-        val cechLatLng = LatLng(cechLat, cechLng)
-        cechMarker = map.addMarker(MarkerOptions()
-                .position(cechLatLng)
-                .title("CECH")
-                .snippet("College of Education, Criminal Justice, Human Services, and IT")
-        )
+        cechMarker = getCechMarker()
     }
+
     override fun onInfoWindowClick(p0: Marker?) {
         if (p0 == tucMarker){
-            val intent = Intent(this, TUC_Info::class.java)
+            val intent = Intent(this, InfoActivity::class.java)
             startActivity(intent)
         }
         else if (p0 == cechMarker){
-            val intent = Intent(this, CECH_Info::class.java)
+            val intent = Intent(this, InfoActivity::class.java)
```

Comment:
Now that both these activities are the same, you can give this `p0` variable to your ViewModel, and let the ViewModel find the DTO for TUC or CECH depending on which marker was passed. The `InfoActivity` class can then use this DTO to populate itself with data. I did not make that change here since you don't have a ViewModel class yet.